google-cloud-bigquery/lib/google/cloud/bigquery/job.rb

```diff
@@ -239,12 +244,16 @@ def reload!
         #                               format: "json"
         #   extract_job.wait_until_done!
         #   extract_job.done? #=> true
-        def wait_until_done!
-          backoff = ->(retries) { sleep 2 * retries + 5 }
+        def wait_until_done! min: 5, max: Float::INFINITY
```

Comment:
I like all your suggestions @blowmage. And I'm really glad @Altech opened this PR.

@Altech If this is going beyond what you had time for, you can also close this PR and reference it in a new issue, and we will do the work. We really appreciate your effort so far!